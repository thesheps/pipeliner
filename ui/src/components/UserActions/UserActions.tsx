import React from "react";

import { RegisterContainer } from "../Register";
import { SignInContainer } from "../SignIn";

export interface UserActionsProps {
  isSignedIn: boolean;
}

export const UserActions = ({ isSignedIn }: UserActionsProps) => (
  <>
    {!isSignedIn && (
      <>
        <SignInContainer />
        <RegisterContainer />
      </>
    )}
  </>
);
