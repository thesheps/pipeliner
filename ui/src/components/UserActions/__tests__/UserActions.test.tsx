import React from "react";
import { render } from "@testing-library/react";

import { UserActions } from "../UserActions";

describe("UserActions", () => {
  it("renders correctly", () => {
    const { container } = render(
      <UserActions registerUser={jest.fn()} isSignedIn={false} />
    );
    
    expect(container).toMatchSnapshot();
  });

  it("renders register button when not signed-in", () => {
    const { queryByTestId } = render(
      <UserActions registerUser={jest.fn()} isSignedIn={false} />
    );

    expect(queryByTestId("register-dialog-button")).toBeTruthy();
  });

  it("does not render register button when signed-in", () => {
    const { queryByTestId } = render(
      <UserActions registerUser={jest.fn()} isSignedIn={true} />
    );

    expect(queryByTestId("register-dialog-button")).toBeNull();
  });
});
