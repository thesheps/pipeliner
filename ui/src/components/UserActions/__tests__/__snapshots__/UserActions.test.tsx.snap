// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserActions renders correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    data-testid="signIn-dialog-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      SignIn
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    data-testid="register-dialog-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Register
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
